gateway/hail.nginx.conf.in

```diff
@@ -158,3 +161,113 @@ server {
+        internal;
+
+        resolver kube-dns.kube-system.svc.cluster.local;
+        proxy_pass http://auth-gateway.default.svc.cluster.local/verify/$auth_request_uri;
+    }
+
+    location = /auth-notebook {
+        internal;
+
+        resolver kube-dns.kube-system.svc.cluster.local;
+        proxy_pass http://notebook-api.default.svc.cluster.local/api/verify/$svc_name/$auth_request_uri;
+    }
+
+    location / {
+        if ($request_method = 'OPTIONS') {
+            add_header 'Access-Control-Allow-Origin' '*';
+            add_header 'Access-Control-Allow-Credentials' 'true';
+            add_header 'Access-Control-Allow-Methods' 'GET, POST, DELETE, OPTIONS';
+            add_header 'Access-Control-Allow-Headers' 'Authorization,Keep-Alive,User-Agent,X-Requested-With,Cache-Control,Content-Type';
+            add_header 'Access-Control-Max-Age' 86400;
+            add_header 'Content-Type' 'text/plain charset=UTF-8';
+            add_header 'Content-Length' 0;
+            return 204;
+            break;
+        }
+
+        set $auth_request_uri "$is_args$args";
+        auth_request /auth;
+        auth_request_set $auth_user $upstream_http_user;
+        auth_request_set $auth_scope $upstream_http_scope;
+        proxy_pass http://notebook-api;
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $http_host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+        proxy_set_header User $auth_user;
+        proxy_set_header Scope $auth_scope;
+
+        proxy_http_version 1.1;
+        proxy_read_timeout 86400;
+        proxy_buffering off;
+    }
+
+    location ~ /instance/([^/]+)/(.*) {
+        set $svc_name $1;
+        #set $auth_request_uri "$is_args$args";
+        #auth_request /auth-notebook;
+
+        resolver kube-dns.kube-system.svc.cluster.local;
+        proxy_pass http://$svc_name.default.svc.cluster.local$request_uri;
+
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $http_host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+        proxy_http_version 1.1;
+        proxy_redirect off;
+        proxy_buffering off;
+        proxy_read_timeout 86400;
+        proxy_connect_timeout 10s;
+    }
+
+    add_header 'Access-Control-Allow-Origin' '*';
+    add_header 'Access-Control-Allow-Credentials' 'true';
+    add_header 'Access-Control-Allow-Methods' 'GET, POST, DELETE, OPTIONS';
+    add_header 'Access-Control-Allow-Headers' 'Authorization,Keep-Alive,User-Agent,Cache-Control,Content-Type';
```

Comment:
I thought we were using Flask-CORS for this.  Is that not the case?

Reply:
We certainly can, mostly wanted to try to separate notebook business logic from public/private  ingress concerns